paderbox/io/download.py

```diff
@@ -1,184 +1,142 @@
+        os.remove(local_file)
+
+    elif local_file.name.endswith('.tar.gz') or local_file.name.endswith('.tar'):
+        mode = "r:gz" if local_file.name.endswith('.tar.gz') else "r"
+        with tarfile.open(local_file, mode) as tar:
+            for i, tar_info in enumerate(tar):
+                target_file = target_dir / tar_info.name
+                if not target_file.exists():
+                    try:
+                        tar.extract(tar_info, target_dir)
+                    except KeyboardInterrupt:
+                        # Delete latest file, since most likely it
+                        # was not extracted fully
+                        if target_file.exists():
+                            os.remove(target_file)
+                        raise
+                elif not exist_ok:
+                    raise FileExistsError(target_file)
+                tar.members = []
+        os.remove(local_file)
+
+
+def download_file_list(file_list, target_dir, extract=True, exist_ok=False, num_workers=1):
     """
     Download file_list to target_dir
 
     Args:
         file_list:
         target_dir:
         exist_ok:
-        logger:
+        extract:
+        num_workers:
 
     Returns:
 
     """
-
     target_dir = Path(target_dir)
     os.makedirs(target_dir, exist_ok=True)
 
-    item_progress = tqdm(
-        file_list, desc="{0: <25s}".format('Download files'),
-        file=sys.stdout, leave=False, ascii=True)
-
-    local_files = list()
-    for remote_file in item_progress:
-        local_files.append(
-            download_file(
-                remote_file,
-                target_dir / Path(remote_file).name,
-                exist_ok=exist_ok
-            )
-        )
-
-    item_progress = tqdm(
-        local_files,
-        desc="{0: <25s}".format('Extract files'),
-        file=sys.stdout,
-        leave=False,
-        ascii=True
-    )
-
-    if logger is not None:
-        logger.info('Starting Extraction')
-    for _id, local_file in enumerate(item_progress):
-        if local_file and local_file.exists():
-            if logger is not None:
-                logger.info(
-                    '  {title:<15s} [{item_id:d}/{total:d}] {package:<30s}'
-                    .format(
-                        title='Extract files ',
-                        item_id=_id,
-                        total=len(item_progress),
-                        package=local_file
-                    )
-                )
-            extract_file(local_file, exist_ok=exist_ok)
+    pbar = tqdm(initial=0, total=len(file_list))
+
+    if isinstance(extract, bool):
+        extract = len(file_list) * [extract]
+    assert len(extract) == len(file_list), (len(extract), len(file_list))
+    if num_workers > 1:
+        with ProcessPoolExecutor(num_workers) as ex:
+            for _ in ex.map(
+                download_file,
+                file_list,
+                [target_dir / Path(f).name.split('?')[0] for f in file_list],
```

Comment:
Could you add a comment, why only name is used and why everything behind the `?` is ignored?
Alternatively, a doctest could also help to understand the motivation.